---
title: "C28125"
ms.custom: na
ms.date: 09/18/2016
ms.prod: visual-studio-dev14
ms.reviewer: na
ms.suite: na
ms.technology: 
  - vs-devops-test
ms.tgt_pltfrm: na
ms.topic: article
ms.assetid: 3f7b8db4-6a64-4480-919e-3f3ceca7f948
caps.latest.revision: 6
translation.priority.ht: 
  - de-de
  - ja-jp
---
# C28125
warning C28125: The function must be called from within a try/except block  
  
 The driver is calling a function that must be called from within a try/except block, such as [ProbeForRead](http://msdn.microsoft.com/library/windows/hardware/ff559876.aspx), [ProbeForWrite](http://msdn.microsoft.com/library/windows/hardware/ff559879.aspx), [MmProbeAndLockPages](http://msdn.microsoft.com/library/windows/hardware/ff554664.aspx).  
  
## Example  
 The following code example generates this warning:  
  
```  
ProbeForRead(addr, len, 4);  
```  
  
 The following code example avoids this warning:  
  
```  
  
__try  
{  
    ProbeForRead(addr, len, 4);  
}  
__except(EXCEPTION_EXECUTE_HANDLER)  
{  
    Status = GetExceptionCode();  
    ... report error status  
}  
```